nuvlaedge/agent/agent.py

```diff
@@ -473,9 +479,15 @@ def _process_jobs(self, jobs: list[NuvlaID]):
         Returns:
             None
         """
+
+        if self.settings.nuvlaedge_exec_jobs_in_agent:
+            coe_engine = self.job_local
+        else:
+            coe_engine = self._coe_engine
```

Comment:
You may want to consider using Protocol from typing package. It's the python style of interfaces and I think would suit very well for this usage